### ISSUE TITLE
DiceKeys.app reported not working on iPhone Safari

### ISSUE THREAD

**Issue by UppaJung** [dicekeys/beta-program#4]

**Comment by drbarrett** [comment]
Try checking Settings. Mine was set to deny, but it works after changing to Ask. However, it does not work if Front Camera is selected in the dicekeys app. It worked fine with the rear camera. I don't know why you'd want to use the front camera anyway...

Settings->Safari->Settings For Websites->Camera

For reference, this was on an iPhone 11 running IOS14.

**Comment by UppaJung** [comment]
Working on adding a feature that will tell you if the problem is that the app isn't allowed to access the cameras.  Should be working by Monday.  Until then, can you confirm that you have given the app access to the camera?

**Comment by ukscone** [comment]
after beating my head against camera usage in various browsers recently (qrcode reading) i've come to hate camera and iOS and various browsers ( https://caniuse.com/?search=camera ) i think it might be a good idea to add a read from image from filesystem function to the app although would need to have some way for the app to "shred" the image after use so it's not left around after reading/using so it's not left lying around if anyone steals/looks in your phone/tablet/...

**Comment by pfmoose** [comment]
I also agree to option to add a image/picture of the dicekey for registration into the DICEKEYS.APP.  I understand ukscone concern with shredding the image.  However it should be reasonable for the user to delete the image as needed.  The image could also be used to backup the key it needed.  

I would securely delete and move the image to an encrypted file location as needed.

**Comment by UppaJung** [comment]
Would you see this feature as necessary after we have a native iOS and Android app?  Native apps should have much more reliable camera access.

**Comment by ukscone** [comment]
if you have native rather than a webapp then it isn't needed. it'd still be a "nice to have one day" but not actually needed. however if push came to shove i'd forego native apps for a webapp as it's more likely to run (browser willing) on anything you throw it at with mostly a single source tree and also run no matter what google and apple do to android and ios in the future although speed of scanning/reading/processing would need improving drastically if it was to stay a webapp as i have time to make a cup of tea between image updates at the moment :)

**Comment by UppaJung** [comment]
For that day in the future when Android and iOS might be distant memories, would you feel more confident knowing the browser feature might still work or if we had a command-line option?

**Comment by ukscone** [comment]
you bring up a good point. you are probably right


[maybe write it in COBOL & release the source then my 7 times grandchildren would still be ok to use the dice :) ]

**Comment by UppaJung** [comment]
Please let me know if the version at https://staging.dicekeys.app/ resolves this issue.
